### PR TITLE
Add contributing document and partially reformat the codebase.

### PR DESCRIPTION
**Pull request by arturdryomov**

### CODE REVIEW

**Review by raniejade — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish
+to make via issue, email, or any other method with the owners of this repository
```
Comment (artem-zinnatullin) [inline CONTRIBUTING.md]: Let's remove

Comment (artem-zinnatullin) [reply CONTRIBUTING.md]: damn apple keyboard

Let's remove "email, or any other method with the owners of this repository"

and only recommend public issues as recommended way of communication

**Comment by artem-zinnatullin** [comment]
I thought you would also commit `.idea/codeStyleSettings` :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (arturdryomov) [reply CONTRIBUTING.md]: Sure, done.

**Review by artem-zinnatullin — APPROVED** [review]

**Pull request merged**